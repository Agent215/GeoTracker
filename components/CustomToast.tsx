import React from 'react';
import Toast from 'react-native-easy-toast';

import Colors from '../constants/Colors';

const CustomToast = React.forwardRef((props, ref) => {
    return (
        <Toast
        ref={ref}
        style={{ backgroundColor: "white", borderRadius: 5 ,zIndex: 1000, width: 200 }}
        position='top'
        positionValue={300}
        fadeInDuration={750}
        fadeOutDuration={1000}
        textStyle={{color:'black'}}
        
    />
    )
})

export default CustomToast;