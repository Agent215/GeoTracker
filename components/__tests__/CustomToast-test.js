import * as React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import CustomToast from '../../components/CustomToast';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
    return {
        ...jest.requireActual('@react-navigation/native'),
        useNavigation: () => ({
            navigate: mockedNavigate,
        }),
    };
});

enzyme.configure({ adapter: new Adapter() });
// mock of a prop to pass customToast
const mockProp = "";


describe('Testing customToast  Component', () => {
    const initialState = { output: "null" }
    const mockStore = configureStore()
    let store
    /*
    Check disaster card against snapshot. run jest -u to update snapshot
    */
    it(`customToast renders correctly`, () => {
        store = mockStore(initialState)

        const comp = <Provider store={store}><CustomToast event={mockProp} /></Provider>
        const tree = renderer.create(comp).toJSON();

        expect(tree).toMatchSnapshot();
    });


})
