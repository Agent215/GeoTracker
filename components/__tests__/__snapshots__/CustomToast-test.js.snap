// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing customToast  Component customToast renders correctly 1`] = `null`;
